bumpversion/functions.py

```diff
@@ -15,13 +15,13 @@ class NumericFunction(object):
     considered (e.g. 'r3-001' --> 'r4-001').
     """
 
-    FIRST_NUMERIC = re.compile("([^\d]*)(\d+)(.*)")
+    FIRST_NUMERIC = re.compile(r"([^\d]*)(\d+)(.*)")
 
     def __init__(self, first_value=None):
 
         if first_value is not None:
             try:
-                part_prefix, part_numeric, part_suffix = self.FIRST_NUMERIC.search(
+                _, _, _ = self.FIRST_NUMERIC.search(
```

Comment:
If all are unused, can you just remove the result? Wouldn't it be cleaner to do:
```python
match = self.FIRST_NUMERIC.search(first_value)
if not match:
    raise ValueError("The given first value {} does not contain any digit".format(first_value))
```